src/dpsnotificationservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/dpsnotificationservice/OutputNotificationEndpoint.java

```diff
@@ -21,14 +27,73 @@
 @Endpoint
 public class OutputNotificationEndpoint {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final RabbitTemplate outputNotificationTopicTemplate;
+
+    public OutputNotificationEndpoint(@Qualifier("outputNotificationTopicTemplate")RabbitTemplate outputNotificationTopicTemplate) {
+        this.outputNotificationTopicTemplate = outputNotificationTopicTemplate;
+    }
+
 
     @PayloadRoot(namespace = Keys.NAMESPACE_URI, localPart = Keys.OUTPUT_NOTIFICATION_REQUEST)
     @ResponsePayload
     public OutputNotificationResponse outputNotificationNotification(@RequestPayload OutputNotificationRequest request) {
+
+
         OutputNotificationResponse response = new OutputNotificationResponse();
         OutputNotificationResponse2 response2 = new OutputNotificationResponse2();
-        response2.setRespCode(Keys.OUTPUT_NOTIFICATION_RESPONSE_SUCCESS_CODE);
-        response2.setRespMsg(Keys.OUTPUT_NOTIFICATION_RESPONSE_SUCCESS_MESSAGE);
+
+        if(request.getOutputNotificationRequest() == null) {
+            logger.error("Missing OutputNotificationRequest");
+            response2.setRespCode(Keys.OUTPUT_NOTIFICATION_RESPONSE_ERROR_CODE);
+            response2.setRespMsg("OutputNotificationRequest is required");
+            response.setOutputNotificationResponse(response2);
+            return response;
+        }
+
+        if(StringUtils.isBlank(request.getOutputNotificationRequest().getBusinessAreaCd())) {
+            logger.error("Missing OutputNotificationRequest.getBusinessAreaCd");
+            response2.setRespCode(Keys.OUTPUT_NOTIFICATION_RESPONSE_ERROR_CODE);
+            response2.setRespMsg("OutputNotificationRequest.getBusinessAreaCd is required");
+            response.setOutputNotificationResponse(response2);
+            return response;
+        }
+
+        if(
+                request.getOutputNotificationRequest().getFileList() == null
+                        || request.getOutputNotificationRequest().getFileList() == null
+                        || request.getOutputNotificationRequest().getFileList().getFileId() == null
+                        || request.getOutputNotificationRequest().getFileList().getFileId().isEmpty()) {
+
+            logger.error("No File to be processed");
+            response2.setRespCode(Keys.OUTPUT_NOTIFICATION_RESPONSE_ERROR_CODE);
+            response2.setRespMsg("OutputNotificationRequest.FileList.FileId must contains a least 1 file");
+            response.setOutputNotificationResponse(response2);
+            return response;
+
+        }
+
+        logger.info("Received new notification for {}", request.getOutputNotificationRequest().getBusinessAreaCd());
```

Comment:
is this a debug or an info?

Reply:
yea nvm, prolly an info but I'm not sure if we're doing this elsewhere in this project like this.

I feel like we're not for some reason, please tell me I'm wrong.

Reply:
We actully need to do that for the rest of the project, this is our logs for splunk where we can monitor how many transactions and so on...